module/surveys/helper.js

```diff
@@ -1808,4 +1809,401 @@ module.exports = class SurveysHelper {
+                    if (new Date() > new Date(solutionDocument[0].endDate)) {
+
+                        if (solutionDocument[0].status == messageConstants.common.ACTIVE_STATUS) {
+                            await solutionsHelper.updateSolutionDocument
+                            (
+                                { link : link },
+                                { $set : { status: messageConstants.common.INACTIVE_STATUS } }
+                            )
+                        }
+                        
+                        throw new Error(messageConstants.apiResponses.LINK_IS_EXPIRED)
+                    }
+                }
+ 
+                let surveyDocument = await this.surveyDocuments
+                (
+                    { solutionId: solutionDocument[0]._id,
+                      createdBy: userId },
+                    ["_id"]
+                )
+                
+  
+                let surveyId;
+
+                if (surveyDocument.length > 0) {
+                    surveyId = surveyDocument[0]._id;
+                }
+                else {
+
+                    let createSurveyDocument = await this.createSurveyDocument
+                        (
+                            userId,
+                            solutionDocument[0]
+                            // userOrgDetails[userId]
+                        )
+
+                    if (!createSurveyDocument.success) {
+                        throw new Error(messageConstants.apiResponses.SURVEY_CREATION_FAILED)
+                    }
+
+                    surveyId = createSurveyDocument.data._id;
+                }
+
+                let validateSurvey = await this.validateSurvey
+                (
+                    surveyId,
+                    userId,
+                    version && version === messageConstants.common.VERSION_3 ? false : true
+                )
+    
+                if (!validateSurvey.success) {
+                    return resolve(validateSurvey);
+                }
+                
+                let surveyDetails = await this.detailsV4
+                (
+                    surveyId,
+                    userId,
+                    validateSurvey.data.submissionId,
+                    roleInformation,
+                    token
+                )
+
+                if (!surveyDetails.success) {
+                    return resolve(surveyDetails);
+                }
+
+                return resolve({
+                    success: true,
+                    message: surveyDetails.message,
+                    data: surveyDetails.data
+                });
+
+                
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    static detailsV4(surveyId = "", userId= "", submissionId = "", roleInformation = {}, userToken ="") {
```

Comment:
@swetha-aggidevara are we not using existing functions?

Reply:
Existing function is being used by other api's and this function gets the data from API